### PR TITLE
DCOS-38823: refactor(NodesTable): Add two columns for Mem

### PR DESCRIPTION
**Pull request by GeorgiSTodorov**
Add two columns for memory usage - progress bar and number value.
Closes DCOS-38823

### CODE REVIEW

File: plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx [hunk 1]
```diff
@@ -3,12 +3,18 @@ import Node from "#SRC/js/structs/Node";
...
+import ProgressBar from "#SRC/js/components/ProgressBar";
 
 export function membarRenderer(data: Node): React.ReactNode {
-  // TODO: DCOS-38823
-  return <span>{data.get("used_resources").mem.toString()}</span>;
+  var value = data.get("used_resources").mem.toString();
+
+  return <ProgressBar
+  className="hidden-medium-down"
+  data={[{ value, className: 'color-2' }]}
```
Comment (juliangieseke) [inline plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: there was a small update to the issue, can you please change this to `color-1` to make it purple? 🙇

File: plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx [hunk 2]
```diff
@@ -3,12 +3,18 @@ import Node from "#SRC/js/structs/Node";
...
 import { IWidthArgs as WidthArgs } from "#PLUGINS/nodes/src/js/types/IWidthArgs";
+import ProgressBar from "#SRC/js/components/ProgressBar";
 
 export function membarRenderer(data: Node): React.ReactNode {
-  // TODO: DCOS-38823
-  return <span>{data.get("used_resources").mem.toString()}</span>;
+  var value = data.get("used_resources").mem.toString();
+
+  return <ProgressBar
+  className="hidden-medium-down"
```
Comment (juliangieseke) [inline plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: this isnt applicable anymore (just remove it) - there is a Issue linked in the CPU bar, can you please add is as comment here? 🙇

**Comment by GeorgiSTodorov** [comment]
@juliangieseke I have updated the PR.

File: plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx [hunk 3]
```diff
@@ -3,12 +3,15 @@ import Node from "#SRC/js/structs/Node";
 // TODO: DCOS-39079
 // import { IWidthArgs as WidthArgs } from "@dcos/ui-kit/packages/table/components/Column";
 import { IWidthArgs as WidthArgs } from "#PLUGINS/nodes/src/js/types/IWidthArgs";
+import ProgressBar from "#SRC/js/components/ProgressBar";
 
 export function membarRenderer(data: Node): React.ReactNode {
-  // TODO: DCOS-38823
-  return <span>{data.get("used_resources").mem.toString()}</span>;
+  var value = data.get("used_resources").mem.toString();
```
Comment (juliangieseke) [inline plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: AFAIK `toString()` isnt necessary here, ProgressBar accepts numbers 👍 (also the `value` variable isnt necessary - see CPUBar for reference)

Comment (DanielMSchmidt) [inline plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: let's use const here 👍

Comment (GeorgiSTodorov) [reply plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: I was thinking of following Julian's suggestion of removing the variable altogether.

Comment (DanielMSchmidt) [reply plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: Also okay :)

File: plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx [hunk 4]
```diff
@@ -3,12 +3,18 @@ import Node from "#SRC/js/structs/Node";
...
+      total={100}
+    />
+  );
 }
+
 export function membarSizer(args: WidthArgs): number {
-  // TODO: DCOS-38823
-  return Math.max(100, args.width / args.totalColumns);
+  // TODO: DCOS-39147
+  return Math.min(60, Math.max(60, args.width / args.totalColumns));
```
Comment (DanielMSchmidt) [inline plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: Isn't this the same as saying x is 60 <= x <= 60? Wouldn't the `Math.max` be enough if you want to express that it has at least the width 60?

Comment (GeorgiSTodorov) [reply plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: I was thinking that I could just return 60, but it was done with Math.min and Math.max in the CPU column, so there must be some reason for that.

Comment (GeorgiSTodorov) [reply plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: https://jira.mesosphere.com/browse/DCOS-39147 - the jira related to this

Comment (DanielMSchmidt) [reply plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: Oh no, that must have slipped through while reviewing 🙈 @juliangieseke What is your stance on this, should we just return 60 or what?

Comment (juliangieseke) [reply plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: hm, yep 🤷‍♂️  but lets not change this here please, there is a follow-up which touches these anyway (see linked todo ticket)

Comment (juliangieseke) [reply plugins/nodes/src/js/columns/NodesTableMemBarColumn.tsx]: updated description of DCOS-39147

File: plugins/nodes/src/js/columns/NodesTableMemColumn.tsx [hunk 5]
```diff
@@ -7,11 +7,10 @@ import { IWidthArgs as WidthArgs } from "#PLUGINS/nodes/src/js/types/IWidthArgs"
 import { SortDirection } from "plugins/nodes/src/js/types/SortDirection";
 
 export function memRenderer(data: Node): React.ReactNode {
-  // TODO: DCOS-38823
-  return <span>{data.get("used_resources").mem.toString()}</span>;
+  return <span>{data.get("used_resources").mem}%</span>;
```
Comment (juliangieseke) [inline plugins/nodes/src/js/columns/NodesTableMemColumn.tsx]: wait, doesnt this have to be percentge?

Comment (juliangieseke) [reply plugins/nodes/src/js/columns/NodesTableMemColumn.tsx]: `return `${data.getUsageStats("mem").percentage}%`;`

**Comment by GeorgiSTodorov** [comment]
I've updated the values to use percentage.

**Pull request merged**